Mage.Sets/src/mage/cards/k/KingNarfisBetrayal.java

```diff
@@ -0,0 +1,154 @@
+
+    static {
+        filter.add(Predicates.or(
+                CardType.CREATURE.getPredicate(),
+                CardType.PLANESWALKER.getPredicate()
+        ));
+    }
+
+    KingNarfisBetrayalFirstEffect() {
+        super(Benefit);
+        staticText = "Each player mills four cards. You may exile up to one creature or planeswalker " +
+                "card from each graveyard. ";
+    }
+
+    private KingNarfisBetrayalFirstEffect(final KingNarfisBetrayalFirstEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public KingNarfisBetrayalFirstEffect copy() {
+        return new KingNarfisBetrayalFirstEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        UUID controllerId = source.getControllerId();
+        Player controller = game.getPlayer(controllerId);
+        if (controller == null) {
+            return false;
+        }
+        millEffect.apply(game, source);
+
+        //Controller graveyard
+        if (controller.getGraveyard().count(filter, game) != 0) {
+            if (controller.chooseUse(outcome, "Exile a creature or planeswalker card from your graveyard?", source, game)) {
+                TargetCard target = new TargetCardInYourGraveyard(filter);
+                target.setNotTarget(true);
+                if (controller.chooseTarget(outcome, controller.getGraveyard(), target, source, game)) {
+                    controller.moveCardsToExile(game.getCard(target.getFirstTarget()), source, game, true, CardUtil.getCardExileZoneId(game, source), CardUtil.createObjectRealtedWindowTitle(source, game, null));
+                }
+            }
+        }
+
+        //Each opponent's graveyard
+        for (UUID opponentId : game.getOpponents(controllerId)) {
+            Player opponent = game.getPlayer(opponentId);
+
+            if (opponent.getGraveyard().count(filter, game) != 0) {
+                if (controller.chooseUse(outcome, "Exile a creature or planeswalker card from " + opponent.getName() + "'s graveyard?", source, game)) {
+                    TargetCard target = new TargetCardInOpponentsGraveyard(1, 1, filter, true);
+                    target.setNotTarget(true);
+                    if (controller.chooseTarget(outcome, opponent.getGraveyard(), target, source, game)) {
+                        controller.moveCardsToExile(game.getCard(target.getFirstTarget()), source, game, true, CardUtil.getCardExileZoneId(game, source), CardUtil.createObjectRealtedWindowTitle(source, game, null));
+                    }
+                }
+            }
+        }
+
+        return true;
+    }
+}
+
+class KingNarfisBetrayalSecondEffect extends OneShotEffect {
+
+
+    public KingNarfisBetrayalSecondEffect() {
+        super(Outcome.Benefit);
+        this.staticText = " Until end of turn, you may cast spells from among cards exiled with King Narfi’s Betrayal," +
```

Comment:
I'm guessing you copied this line from Scryfall.  It's not using the normal apostrophe character so it's giving a test error.

Reply:
Thank you very much. I just saw that the tests failed because of this.
I am currently trying to fix the interaction with adventure cards and it seems that I can only cast the creature part with Thief of Sanity. Can't send Smitten Swordmaster or Bonecrusher Giant on an adventure from exile.